includes/core/classes/class-setup.php

```diff
@@ -218,6 +219,58 @@ public function deactivate_gatherpress_plugin(): void {
 		flush_rewrite_rules();
 	}
 
+	/**
+	 * Uninstalls the GatherPress plugin.
+	 *
+	 * This method handles the uninstallation of the GatherPress plugin, ensuring that all
+	 * options and custom database tables created by the plugin are removed. It is designed
+	 * to handle both single-site and multisite WordPress installations.
+	 *
+	 * When the plugin is uninstalled, this method performs the following actions:
+	 * - Deletes the plugin-specific option from the database.
+	 * - Deletes the site-specific option in a multisite installation.
+	 * - Drops the custom database table created by the plugin.
+	 * - In a multisite network, loops through all sites to drop the custom table for each site.
+	 *
+	 * Note: This method is registered as the uninstall hook using `register_uninstall_hook`.
+	 * It ensures that the script is called from within WordPress by checking the `WP_UNINSTALL_PLUGIN` constant.
+	 *
+	 * @since 1.0.0
+	 *
+	 * @global wpdb $wpdb WordPress database abstraction object.
+	 *
+	 * @return void
+	 */
+	public function uninstall_gatherpress_plugin(): void {
+		// Ensure the uninstallation script is called from within WordPress.
+		if( ! defined( 'WP_UNINSTALL_PLUGIN' ) ) {
+			die;
+		}
+
+		global $wpdb;
+
+		// Delete Options.
+		$option_name = 'wporg_option';
```

Comment:
This was just an example. We need to remove things we store in the options table (GatherPress settings). In addition to deleting options, should delete user meta, taxonomies, events, venues, etc. Some things like post meta will be deleted automatically with the posts. We might want to delete things in options table by looking for anything that prepends with `gatherpress_`. What do you think @carstingaxion?

Reply:
Take a look at https://github.com/stephenharris/Event-Organiser/blob/65c4e3b0f40a601bebf7d5fc2203da02bbb4d7e9/includes/event-organiser-install.php#L404 as a nice example.

Reply:
What I especially liked about the linked implementation was:

- [ ] additional check if plugin is network-enabled 
- [ ] deletion of user_meta

In addition:

For the multiple options I suggest an array function like `array_walk` calling `delete_option` on the list of all options and transients as well.